internal/provider/config_data_source.go

```diff
@@ -118,6 +117,9 @@ func (d *ConfigDataSource) Read(ctx context.Context, req datasource.ReadRequest,
 	if len(ic.Archs) == 0 {
 		if len(d.popts.archs) != 0 {
 			ic.Archs = apkotypes.ParseArchitectures(d.popts.archs)
+		} else {
+			// Default to host arch when provider and config data source don't specify any.
+			ic.Archs = apkotypes.ParseArchitectures([]string{"host"})
```

Comment:
I think elsewhere we default to all architectures 🤔 

in `fromImageData`:
```go
	} else {
		// If neither is specified, build for all architectures!
		bc.ImageConfiguration.Archs = types.AllArchs
	}

```

Reply:
@joshrwolf do you want to clean this up?

Reply:
I matched the default to all archs, ptal